pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/ZkBasedTableRebalanceObserver.java

```diff
@@ -98,10 +117,9 @@ private void updateOnStart(Map<String, Map<String, String>> currentState,
 
   @Override
   public void onSuccess(String msg) {
-    Preconditions.checkState(_tableRebalanceProgressStats.getStatus() != RebalanceResult.Status.DONE.toString(),
+    Preconditions.checkState(!RebalanceResult.Status.DONE.toString().equals(_tableRebalanceProgressStats.getStatus()),
```

Comment:
Good catch! IMO a better fix is to directly put `Status` instead of `String` in the `_tableRebalanceProgressStats`

Reply:
sgtm